source/common/upstream/cluster_manager_impl.cc

```diff
@@ -1355,9 +1360,6 @@ ClusterManagerImpl::ClusterInitializationObject::ClusterInitializationObject(
     : per_priority_state_(per_priority_state), cluster_info_(std::move(cluster_info)),
       load_balancer_factory_(load_balancer_factory), cross_priority_host_map_(map) {
 
-  ASSERT(cluster_info_->type() == envoy::config::cluster::v3::Cluster::EDS,
-         fmt::format("Using merge constructor on possible non-mergable cluster of type {}",
-                     cluster_info_->type()));
```

Comment:
In the tests, we may call the `updateHosts` of static cluster manually, we should ensure updates in that case also be merged correctly.

Reply:
I wonder if it will be possible to add an assertion that captures the comment above (either EDS or `updateHosts` in tests)?
Not sure if there is, but it may be useful to capture this to avoid wrong calls.

Reply:
Although from technical point, anything is possible, but it's hard 🤣